src/include/cell_inline.h

```diff
@@ -332,6 +332,9 @@ __wt_cell_pack_value_match(
         if (rle) /* Skip RLE */
             WT_RET(__wt_vunpack_uint(&a, 0, &v));
         WT_RET(__wt_vunpack_uint(&a, 0, &alen)); /* Length */
+        /* We need to adjust the size to ignore the validity window or RLE information. */
```

Comment:
Why we need to do that?

Reply:
[Here](https://github.com/wiredtiger/wiredtiger/blob/1d24ec46e33c195737202c5aa27f13485b1786f6/src/include/cell_inline.h#L271), we already adjust the size in `__wt_cell_pack_value` depending on the presence of time window/rle info. I figured that sometimes, the cells would have the same value but not the same length because of the RLE field being present. Since we want to skip the RLE for comparison, the size should be adjusted.

Reply:
I think you should expand the comment to explain this.

Reply:
Are you sure the size passed in this function hasn't considered the adjustment?

Reply:
> Are you sure the size passed in this function hasn't considered the adjustment?

Yes I am sure, with the reproducer I can see that we are comparing the same value but `page_cell` and `val_cell` have a different size because one has a `RLE >= 2`.

Reply:
I wonder if we are missing the length adjustment somewhere in `__wt_cell_unpack_safe`.

Reply:
Can we add an assert in the end that if the compare is the same the length should all be same?

Reply:
We only compare the values if the sizes are the same, see [here](https://github.com/wiredtiger/wiredtiger/blob/f91ceec86ebb104b170a8106577d7246c6049c54/src/include/cell_inline.h#L367-L368).

Reply:
@quchenhao and I had a discussion on zoom and it would be better to keep the size adjustment in the packing/unpacking code. However, it does make sense to reduce the size when writing to disk if we are going to save space. But why would we increase the size when reading from disk? The size should remain the same that we used when writing to disk.

The issues comes when:

1. we have written a cell on a page that has no RLE information (we have reduced the size)
2. we want to write a new cell on that page that contains RLE information. When comparing them, we want to make sure to skip that RLE as it should not be taken into account when comparing the values

If we do adjust the size when unpacking, we should be fine with the comparison. If we adjust the size when comparing, the code seems out of pace?